### PR TITLE
Handle podspec file

### PR DESCRIPTION
**Pull request by rpotter12**
- add code to handle podspec file.
- update scripts to detect all types of dependencies from the file.

### CODE REVIEW

**Review by steven-esser — CHANGES_REQUESTED** [review]
@rpotter12 Are the various formatting changes done by you or by some formatter tool?

Also, we probably want to Add or Update the README file to explain the reason for the fork and what exactly is different than `gemfileparser`

**Comment by rpotter12** [comment]
> @rpotter12 Are the various formatting changes done by you or by some formatter tool?

@majurg All the formatting changes are done by me only. :)

> Also, we probably want to Add or Update the README file to explain the reason for the fork and what exactly is different than `gemfileparser`

Okay. I will update the README file soon :)

**Review by steven-esser — APPROVED** [review]
@rpotter12 Since we fork, I am not sure we want to have stuff like syntax changes. I think in general, formatting changes like this are a good thing. I am more concerned about having a clean diff in case we want to merge this in upstream in the future. 

@pombredanne What is your take here?

@rpotter12 Other than that particular minor issue, this looks good to me.

File: gemfileparser/__init__.py
Comment (pombredanne) [inline gemfileparser/__init__.py:40]: Why use empty strings instead of None?

File: gemfileparser/__init__.py
Comment (pombredanne) [inline gemfileparser/__init__.py:44]: I guess you removed these attributes because they are either not in the spec or seldom used or parsable?
platform is still a thing though rarely there alright https://guides.rubygems.org/specification-reference/#platform=
Can you confirm?

I never saw a group in a spec so far... did you?

File: gemfileparser/__init__.py
Comment (pombredanne) [inline gemfileparser/__init__.py:108]: Please use the imperative style for docstrings.

File: gemfileparser/__init__.py
Comment (pombredanne) [inline gemfileparser/__init__.py:122]: Can you remind my the benefits of using a CSV module here?

File: setup.py
Comment (pombredanne) [inline setup.py:9]: Please keep the original author name there. Or something like `Rohit Potter for nexB based on original work of Balasankar C' ... but in all cases we want Balasankar to get proper credits

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thank you! I only have a few nits for your review... but most and foremost we need to have tests for these before we can merge.

File: gemfileparser/__init__.py
Comment (rpotter12) [reply gemfileparser/__init__.py:44]: I think they are used in `gemfile` files. `gemfileparser` is for both gemspec and gemfile but I think @balankarc have implement these in another way so he removed this in the original repository :)

File: gemfileparser/__init__.py
Comment (rpotter12) [reply gemfileparser/__init__.py:108]: I will change it soon :)

File: gemfileparser/__init__.py
Comment (rpotter12) [reply gemfileparser/__init__.py:122]: I think @balasankarc used this CSV module here because dependency contains name and different version numbers and they are seperated by comma and they have to split by comma and if we use something else then to remove extra character the code will become more complex. CSV module might have make this easy to remove extra characters. I guess this might be the only good solution to use CSV module here.

File: gemfileparser/__init__.py
Comment (rpotter12) [reply gemfileparser/__init__.py:40]: I think the repo which we have forked is some commits behind. Using empty strings here are the original changes in the upstream repo. And may be this is helpful in extracting data from `gemfile` files.

File: setup.py
Comment (rpotter12) [reply setup.py:9]: Okh :)
I will change this in new commits.

**Comment by pombredanne** [comment]
Never mind the auto close.... that's Github playing tricks on us.

**Comment by balasankarc** [comment]
@pombredanne @rpotter12 Wanted to open an issue about this, but since issue tracker is not enabled here I am "hijacking" this thread. :grinning: Sorry about that.

If you are interested, I am more than happy to move `gemfileparser` into an organization and add you to it, so instead of maintaining a fork you can use the upstream repository itself. I am not sure about how much time I might be able to dedicate in maintaining it though, but if the project is of use to you, I think it is only right that you get to be an upstream contributors/co-authors.

File: setup.py
Comment (balasankarc) [reply setup.py:9]: In case we get to collaborate and move development upstream as suggested in https://github.com/nexB/gemfileparser2/pull/2#issuecomment-663682411, this can definitely change to an AUTHOR file based logic so that every contributor gets credit. :slightly_smiling_face:

**Comment by rpotter12** [comment]
@balasankarc Thank you for your reply :)
`gemfileparser` is very helpful in the `scancode-toolkit` software. I think you can transfer ownership to @pombredanne or make us maintainer of that repo if you are not able to dedicate much time on it. We also need to update the Pypi version also because it is having the very old version `gemfileparser` and needs to be updated regularly. 

Thank you so much for you reply :)

**Comment by pombredanne** [comment]
@balasankarc re

> if you are interested, I am more than happy to move gemfileparser into an organization and add you to it, so instead of maintaining a fork you can use the upstream repository itself. I am not sure about how much time I might be able to dedicate in maintaining it though, but if the project is of use to you, I think it is only right that you get to be an upstream contributors/co-authors.


That would be great! 

We can either create a new org or we can host it under nexB org as this is meant for use directly in https://github.com/nexB/scancode-toolkit which is in the same org. (ScanCode is a tool that you should consider for use in Gitlab ... even if this is Python and not Ruby as this is the best tool out there for license detection ;) 

Note that I would like also to fold in that gemfileparser library the support we have for Gemfile.lock in here https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/gemfile_lock.py

So your call in anycase.
AFAIK the latest code is in your gitlab repo and not in Github ;).

**Comment by balasankarc** [comment]
1. Moved upstream to https://github.com/gemfileparser/gemfileparser
1. Synced GitHub with GitLab. Now both of them should have same code.
1. Added @pombredanne as an owner to https://github.com/gemfileparser so the needful can be done without depending on me.

@pombredanne If you have a PyPi profile, please share your email address and I can add you as a Maintainer to the package there too so you can update the package.

I will update README and add an AUTHOR file and everything in upstream repo soon.

**Comment by pombredanne** [comment]
@balasankarc you wrote:
> Moved upstream to https://github.com/gemfileparser/gemfileparser
> Synced GitHub with GitLab. Now both of them should have same code.
> Added @pombredanne as an owner to https://github.com/gemfileparser so the needful can be done without depending on me.
>
> @pombredanne If you have a PyPi profile, please share your email address and I can add you as a Maintainer to the package there too so you can update the package.
>
> I will update README and add an AUTHOR file and everything in upstream repo soon.

That's great! Thank you: I am pombredanne @ PyPI too.
@rpotter12 can you move your PR to
https://github.com/gemfileparser/gemfileparser  then?

-- 
Cordially
Philippe Ombredanne

**Comment by pombredanne** [comment]
Closing these as we eventually used and pushed all these upstream!

**Pull request closed without merging**
